packages/launcher/src/renderer/launcher/PermissionsView.js

```diff
@@ -112,49 +115,90 @@ export default class PermissionsView extends Component<Props, State> {
   onPressDone = () => {
     const { permissions } = this.props
     const { permissionsSettings } = this.state
-
-    let invalid = false
-    if (permissions.required != null) {
-      const { HTTPS_REQUEST, ...others } = permissions.required
-      // Check expected values for basic keys (boolean)
-      invalid = Object.keys(others).some(
-        key => permissionsSettings[key] !== true,
-      )
-      if (invalid === false && HTTPS_REQUEST != null) {
-        // Check special HTTPS_REQUEST case
-        invalid = HTTPS_REQUEST.some(
-          domain => permissionsSettings.HTTPS_REQUEST[domain] !== true,
-        )
-      }
-    }
-
-    if (invalid) {
-      this.setState({ failedValidation: true })
-    } else {
-      this.props.onSubmit(formatSettings(permissionsSettings))
+    const settings = formatSettings(permissionsSettings)
+    const requiredDomains = permissions.required['HTTPS_REQUEST']
+    if (requiredDomains && requiredDomains.length) {
+      settings['HTTPS_REQUEST'].granted.push(...requiredDomains)
+      delete permissions.required['HTTPS_REQUEST']
     }
+    this.props.onSubmit({ ...settings, ...permissions.required })
   }
 
-  onToggle = (
+  onSetPermissionGrant = (
     key: PermissionKey,
-    granted: PermissionGranted,
+    granted: ?PermissionGranted,
     option?: string,
   ) => {
     this.setState(({ permissionsSettings }) => {
       if (key === 'HTTPS_REQUEST') {
         if (option == null) {
           return null // Don't update state
         }
-        permissionsSettings.HTTPS_REQUEST[option] = granted
+        if (granted === undefined || granted === null) {
```

Comment:
Rather than checking for `undefined` or `null` with strict equality (`===`), you can use `== null` to achieve the same effect.